components/Toast/index.jsx

```diff
@@ -0,0 +1,26 @@
+import React, { useEffect } from "react";
+
+import { ToastContainer, CloseButton } from "@/design-system/StyledToast";
+
+function Toast({ message, showToast, setShowToast, type }) {
+  useEffect(() => {
+    if (showToast) {
+      const timer = setTimeout(() => {
+        setShowToast(false);
+      }, 3000); // i will adjust the duration as needed later
+
+      return () => clearTimeout(timer);
+    }
+  }, [showToast, setShowToast]);
+
+  return (
+    <ToastContainer showToast={showToast} type={type}>
```

Comment:
when running your code locally I got an error in my terminal
`styled-components: it looks like an unknown prop "showToast" is being sent through to the DOM, which will likely trigger a React console error.`

You can get rid of this message by prefixing the showToast prop with a dollar sign.

```suggestion
    <ToastContainer $showToast={showToast} type={type}>
```

(You will also need to do this in your ToastContainer component)